lib/compile.js

```diff
@@ -57,26 +57,37 @@ function Compile(structure, opts) {
     }
 
     function buildToken(ruleName, token) {
+        if (typeof token === 'string') {
+            if (token === 'null') {
+                return null;
+            }
+            return token;
+        }
+
+        if (token instanceof RegExp) {
+            return token;
+        }
+
         if (token.literal) {
-            var str = token.literal;
-            if (str.length > 1) {
-                return buildStringToken(ruleName, token);
-            } else if (str.length === 1) {
-                return {
-                    literal: str
-                };
+            if (!token.literal ||
```

Comment:
A little confused by this check. Is this trying to check for the empty string? If so, wouldn't `token.literal === ''` suffice?